File: src/app/core/services/game.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { Game, GamePrice, GameAvailability } from '../../models/game.model';

@Injectable({
  providedIn: 'root'
})
export class GameService {
  private apiService = inject(ApiService);

  getAllGames(): Observable<Game[]> {
    return this.apiService.get<Game[]>('/games');
  }

  getActiveGames(): Observable<Game[]> {
    return this.apiService.get<Game[]>('/games/active');
  }

  getGameById(id: string): Observable<Game> {
    return this.apiService.get<Game>(`/games/${id}`);
  }

  createGame(game: Partial<Game>): Observable<Game> {
    return this.apiService.post<Game>('/games', game);
  }

  updateGame(id: string, game: Partial<Game>): Observable<Game> {
    return this.apiService.put<Game>(`/games/${id}`, game);
  }

  deleteGame(id: string): Observable<void> {
    return this.apiService.delete<void>(`/games/${id}`);
  }
}

File: backend/src/main/java/com/vrroom/service/GameService.java
package com.vrroom.service;

import com.vrroom.dto.GameDTO;
import java.util.List;

public interface GameService
{

    List<GameDTO> getAllGames();

    List<GameDTO> getActiveGames();

    List<GameDTO> getGamesByDifficulty(Integer difficulty);

    GameDTO getGameById(String id);

    GameDTO getGameByCode(String id);

    GameDTO createGame(GameDTO gameDTO);

    GameDTO updateGame(String id, GameDTO gameDTO);

    void deleteGame(String id);

    void deactivateGame(String id);
}
